### PR TITLE
Xtce generator updates

### PR DESCRIPTION
**Pull request by tech3371**
# Change Summary

## Overview
Refactored XTCE generator code to work with different dataType and remove needing to specify specific mnemonic.

## New Files
- Create `xtce_generator_template.py` that we can use to generate any number of XTCE for any instrument.

## Deleted Files
- Deleted files from `imap_processing/codice/L0` folder which are being replaced by `xtce_generator_template.py`

## Updated Files
- moved xml files from `imap_processing/packet_definitions` into their instrument folder
- tools/xtce_generation/ccsds_header_xtce_generator.py. Updated `unint` to `UINT`
- tools/xtce_generation/telemetry_generator.py
   - main refactor in this PR happened in this file

### CODE REVIEW

File: tools/xtce_generation/telemetry_generator.py
Comment (maxinelasp) [inline tools/xtce_generation/telemetry_generator.py]: Comment needs updating

File: tools/xtce_generation/xtce_generator_template.py
Comment (maxinelasp) [inline tools/xtce_generation/xtce_generator_template.py:6]: Nice! This example is great!

File: tools/xtce_generation/telemetry_generator.py
Comment (maxinelasp) [inline tools/xtce_generation/telemetry_generator.py]: If someone wants to edit the packet definitions before calling TelemetryGenerator, then moving this parse step inside the class makes it harder to do that. For example, the GLOWS sheet contains a few extra columns which are not part of the packet - I removed them in Pandas before calling TelemetryGenerator. 

I think this is fine, because you can always edit the excel sheet, but I do want to call out that loss of flexibility by doing the parsing inside the class. Depending on how much you need to change the excel document, it could add some extra effort for processing new versions of the sheet.

File: tools/xtce_generation/telemetry_generator.py
Comment (maxinelasp) [inline tools/xtce_generation/telemetry_generator.py]: Is SHCOARSE from the CCSDS header? I don't see that name in GLOWS excel document, but there is a "CCSDS packet header coarse time" which might be what that corresponds to?

**Review by maxinelasp — COMMENTED** [review]
This looks great! I will be testing locally with the GLOWS packet to see how the output looks and how it works. Thanks for the updates! I'll approve once I test it out locally, but just had a few comments first :)

File: tools/xtce_generation/telemetry_generator.py
Comment (tech3371) [reply tools/xtce_generation/telemetry_generator.py]: yes!

File: tools/xtce_generation/telemetry_generator.py
Comment (tech3371) [reply tools/xtce_generation/telemetry_generator.py]: `SHCOARSE` is also know as `MET`.

File: tools/xtce_generation/telemetry_generator.py
Comment (tech3371) [reply tools/xtce_generation/telemetry_generator.py]: Can you give an example? 

Removing column should not effect it. We mainly use these column in generator - `lengthInBits`, `dataType`, `mnemonic`, `longDescription`, `shortDescription`. If those column contains, it should still work.

File: tools/xtce_generation/telemetry_generator.py
Comment (maxinelasp) [reply tools/xtce_generation/telemetry_generator.py]: Ah yes, that's what it's called. Can you update the comment to make that clear?

File: tools/xtce_generation/telemetry_generator.py
Comment (maxinelasp) [reply tools/xtce_generation/telemetry_generator.py]: The GLOWS sheet has a column which sums up all the bits - presumably as a check for the size of the packet. If I wanted to remove that programmatically, I wouldn't be able to, because the excel file is parsed inside this class instead of the class having the packet passed in as a dataframe. 

```
xls = pd.ExcelFile(excel_file)
    packet_df = xls.parse(packet_name)

    # Clean up packet
    packet_df.drop(
        labels=packet_df.columns[packet_df.columns.str.contains("Unnamed")],
        axis=1,
        inplace=True,
    )
    packet_df.dropna(subset="packetName", inplace=True)

    generator = GlowsTelemetryGenerator(packet_name, excel_file, apid, pkt=packet_df)
```

Comment (maxinelasp) [reply tools/xtce_generation/telemetry_generator.py]: it's hard to explain, and barely matters 😅 but this bit of code just chops off the unnamed columns and cleans up the sheet.

File: imap_processing/codice/packet_definitions/P_COD_HI_OMNI_SPECIES_COUNTS.xml
Comment (GFMoraga) [inline imap_processing/codice/packet_definitions/P_COD_HI_OMNI_SPECIES_COUNTS.xml:179]: This looks great!

File: tools/xtce_generation/xtce_generator_template.py
Comment (GFMoraga) [inline tools/xtce_generation/xtce_generator_template.py:48]: This looks great, but I will test it here in a bit!

**Comment by GFMoraga** [comment]
I am assuming that we will build documentation for how to use this?

File: tools/xtce_generation/telemetry_generator.py
Comment (tech3371) [inline tools/xtce_generation/telemetry_generator.py]: @maxinelasp If I change it to this, will it help?
```suggestion
         if pkt is None:
            # Read excel sheet
            xls = pd.ExcelFile(path_to_excel_file)
            self.pkt = xls.parse(packet_name)
        else:
            self.pkt = pkt
```

File: tools/xtce_generation/telemetry_generator.py
Comment (tech3371) [reply tools/xtce_generation/telemetry_generator.py]: for sure! I updated it in last commit.

File: tools/xtce_generation/xtce_generator_template.py
Comment (tech3371) [reply tools/xtce_generation/xtce_generator_template.py:48]: In here, you can add other definitions you need to convert by adding packet name and appId to the `packet` dict.

File: tools/xtce_generation/telemetry_generator.py
Comment (maxinelasp) [reply tools/xtce_generation/telemetry_generator.py]: Yes!

Comment (tech3371) [reply tools/xtce_generation/telemetry_generator.py]: cool. I will make that change.

**Review by maxinelasp — APPROVED** [review]
Works great! Thanks for this. Can you squash your commits before merging?

**Comment by tech3371** [comment]
pre-commit.ci autofix

**Review by GFMoraga — APPROVED** [review]
Thanks for updating this and having an impromptu wg! It looks great!

**Pull request merged**
